### PR TITLE
Figures and Axes

### PR DESCRIPTION
**Pull request by bbengfort**
This PR adds some API documentation about Yellowbrick's use of Figures and Axes. This documentation straddles the line between what users would find interesting and what is required for developing new visualizers. I placed it in the API section at the end, but am happy to move it about. 

The page is also unfinished, it ends in two questions which I want to generally ask the larger PyData and Python visualization community. Hopefully, in the future, the answer to these questions will allow us to fill in better examples of `make_axes_locateable` or the AxesGrid toolkit.

### CODE REVIEW

File: docs/api/figures.rst [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+- The confusion matrix probability histogram is a grid of axes for each class pair
+- The ICDM has an inset axes that acts as a dynamic legend
+
+Although it would have been easier to simply embed the figure into the visualizer and use a ``GridSpec`` or other layout tool, the focus on ensuring visualizers are individual plots that wrap an Axes has made us bend over backward to adjust the plot inside of the axes area that was originally supplied, primarily by using ``make_axes_locateable``, which is part of the AxesGrid toolkit.
+
+Generally, it appears that the `AxesGrid Toolkit <https://matplotlib.org/mpl_toolkits/axes_grid/users/overview.html#imagegrid>`_ is the right tool for Yellowbrick - many of the examples shown are similar to the things that Yellowbrick is trying to do. However, this package is not fully documented with examples and some helper utilities that would be useful, for example the ``ImageGrid``, still require a ``figure.Figure`` object.
+
+At this point we are left with some important questions about Yellowbrick's development roadmap:
+
+1. Like Seaborn, should YB have to classes of visualizer, one that wraps an axes and one that wraps a figure?
```
Comment (lwgray) [inline docs/api/figures.rst]: These questions should be posted to our listserv, twitter,  
numfocus listserv

Comment (bbengfort) [reply docs/api/figures.rst]: I've added a para at the top that makes this more like an "open letter". Only the change in `docs/matplotlib.rst` is really important; should we include this in our docs to send around to the listservs, or should we just take the draft copy and send it ourselves?

Comment (lwgray) [reply docs/api/figures.rst]: not sure I  understand what you mean "include in docs to send" versus "a draft copy"

Comment (bbengfort) [reply docs/api/figures.rst]: Yep, sorry I can see how that was unclear. I mean should we merge this PR into our docs and send around a link vs do not merge it into our docs and send it directly (thus the PR acts as a draft).

Comment (lwgray) [reply docs/api/figures.rst]: I think it should be merged into the docs and a link sent around

Comment (bbengfort) [reply docs/api/figures.rst]: Ok, in that case - let me know if you need anything for this to be approved and we'll get it merged in!

**Pull request merged**